zones.go

```diff
@@ -16,31 +16,30 @@ type DownloadInfo struct {
 	Filename      string
 }
 
+// DownloadZoneToWriter is analogus to DownloadZone but instead of writing it to a file, it will
+// write it to a provided io.Writer.
+func (c *Client) DownloadZoneToWriter(url string, dest io.Writer) error {
+        resp, err := c.apiRequest(true, "GET", url, nil)
+        if err != nil {
+                return err
+        }
+        defer resp.Body.Close()
+
+        _, err = io.Copy(dest, resp.Body)
+        return err
+}
+
 // DownloadZone provided the zone download URL retrieved from GetLinks() downloads the zone file and
 // saves it to local disk at destinationPath
 func (c *Client) DownloadZone(url, destinationPath string) error {
-	resp, err := c.apiRequest(true, "GET", url, nil)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-
-	// start the file download
-	file, err := os.Create(destinationPath)
-	if err != nil {
-		return err
-	}
-	defer file.Close()
-
-	n, err := io.Copy(file, resp.Body)
-	if err != nil {
-		return err
-	}
-	if n == 0 {
-		return fmt.Errorf("%s was empty", destinationPath)
-	}
```

Comment:
Can you add this empty check into `DownloadZoneToWriter`?

Reply:
I'd argue that an empty response body is not an error. If you look at io.Copy's semantics, or anything in the standard library, an error is more traditionally used to indicate some exceptional condition. io.Copy returns an int64 to inform the user of what was actually copied and doesn't overload an error for that case. I'll make a change that doesn't change the behavior of DownloadZone but makes DownloadZoneToWrite behave more idiomatically.